core/controllers/suggestion.py

```diff
@@ -509,17 +509,17 @@ def put(self, suggestion_id):
         """Handles PUT requests.
 
         Raises:
-            InvalidInputException. The suggestion is already handled.
+            InvalidInputException. The suggestion is already edited after
+                being rejected.
             InvalidInputException. The 'translation_html' parameter is missing.
             InvalidInputException. The 'translation_html' parameter is not a
                 string.
         """
         suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
-        if suggestion.is_handled:
+
+        if suggestion.edited_after_rejecting:
```

Comment:
This is a bit odd. I don't understand why you need this field ... if the contributor tries to resubmit a rejected submission, could you instead check whether (a) the opportunity is still open and (b) there isn't already a submission in review for it, and accept it if both conditions are true and reject it otherwise?

The concern I have about maintaining this field is that its name is kind of unclear, since the rejected submission is not _actually_ being edited and it also seems a bit odd for a suggestion to contain info about a different model altogether. So I wonder if the check should be framed differently. What do you think?